src/react.tsx

```diff
@@ -23,7 +23,7 @@ export function PreloadTranslations({ loadPath }: PreloadTranslationsProps) {
     ...new Set(
       useMatches()
         .filter((route) => route.handle?.i18n !== undefined)
-        .flatMap((route) => route.handle.i18n as string | string[])
+        .flatMap((route) => route.handle?.i18n as string | string[])
```

Comment:
Even with the filter on the previous line, TypeScript was still reporting possibly undefined for the handle property.

Reply:
Yeah `filter` does not work like that unfortunately. We need to pass a custom type guard directly for it to work as expected

cf. https://www.skovy.dev/blog/typescript-filter-array-with-type-guard